Frontiers_317_delete_products.py

```diff
@@ -0,0 +1,64 @@
+"""
+Working on a script to clean up demo store. delete all products without images.
+"""
+
+import logging
+import os
+from woocommerce import API
+
+logging.basicConfig(level=logging.INFO)
+
+woo_key = 'WOO_KEY'
+woo_secret = 'WOO_SECRET'
+
+def check_env_variables(variable_name):
+    try:
+        os.environ[variable_name]
+    except KeyError as e:
+        error_message = f"The environment variable must be set: {e}"
+        logging.exception(error_message)
+        raise Exception(error_message)
+
+check_env_variables(woo_key)
+check_env_variables(woo_secret)
+
+wcapi = API(
+    url="http://dev.bootcamp.store.supersqa.com",
+    consumer_key=os.getenv(woo_key),
+    consumer_secret=os.getenv(woo_secret),
+    version="wc/v3",
+    timeout=60
+)
+
+page = 1
+total_products = []
+products_to_delete = []
+
+# get all products from db
+while True:
+    r = wcapi.get("products", params={"per_page": 100, 'page': page})
+    products = r.json()
+    page = page + 1
+    if not products:  # no more products
+        break
+
+    # add products into lists. total products and products with no images to be deleted
+    for product in products:
+        total_products.append(product['id'])
+        if not product['images']:
+            products_to_delete.append(product['id'])
+
+# take user input and delete products
+def delete_confirm():
+    logging.info(f"There are {len(total_products)} products currently in your store.")
+    delete = input(f"Are you sure you want to permanently delete {len(products_to_delete)} products? (Yes / N):").lower()
+
```

Comment:
I love how you are confirming with the user. 
But how would this work in Jenkins?